arclightning_backend/src/router.rs

```diff
@@ -164,14 +248,124 @@ impl Router {
         Box::new(response)
     }
 
+    // Checks password at demo screen
+    // If correct, returns serialized access key in the ResponseFuture
+    fn check_password(
+        &mut self,
+        request: Request<Body>,
+        hashed_password: String,
+    ) -> ResponseFuture {
+        let access_key = self.access_key.clone();
+
+        let response = request
+            .into_body()
+            .concat2()
+            .map_err(|err| {
+                io::Error::new(
+                    ErrorKind::Other,
+                    format!("Failed to parse byte string: {}", err),
+                )
+            }).and_then(|body| {
+                serde_json::from_slice(&body).map_err(|err| io::Error::new(ErrorKind::Other, err))
+            }).and_then(move |request_body: PasswordRequest| {
+                let password = request_body.password.clone();
+
+                // 64 random bytes encoded as hex and stored as a string
+                let mut rng = rand::thread_rng();
+                let mut session_token: String = String::new();
+                for _ in 0..64 {
```

Comment:
I would create a byte array and use https://docs.rs/rand/0.6.0/rand/trait.Rng.html#method.fill here, then convert the bytes to a  hex  string with the `hex` crate